src/DI/ConsoleExtension.php

```diff
@@ -153,7 +153,9 @@ public function beforeCompile(): void
 		if ($config->url !== null && $builder->hasDefinition('http.request')) {
 			/** @var ServiceDefinition $httpDef */
 			$httpDef = $builder->getDefinition('http.request');
-			$httpDef->setFactory(Request::class, [new Statement(UrlScript::class, [$config->url])]);
```

Comment:
What about this.

```
$httpDef->setFactory($httpDef->getFactory(), [new Statement(UrlScript::class, [$config->url])]);
```

You must preserver arguments order/position.

Reply:
I am worried it would still break do to possibly different constructor

Reply:
I was thinking about that too, but i didnt like the idea of preserving argument positions. Original idea was cleaner for me.